src/core/internal/retry/index.ts

```diff
@@ -0,0 +1,60 @@
+export interface RetryConfig {
+    //amount of retries
+    retries: number;
+    //wait interval in ms between retries
+    waitInterval: number;
+    //array of conditions which will check if retry should not be applied
+    conditions: SkipRetryCondition[];
+}
+
+export interface SkipRetryCondition {
+    test(error: any): boolean;
+}
+
+export class RetryService {
+    constructor(private config: RetryConfig) {}
+
+    /**
+     *
+     * @param func function to execute (should return {@link promise})
+     */
+    async executeAsyncFunc(func: Function): Promise<any> {
+        for (let i = 0; i < this.config.retries; i++) {
```

Comment:
```suggestion
        for (let i = 0; i <= this.config.retries; i++) {
```

Reply:
Well it’s really depends on how we count retries. My solution was based on that even first (standard) execution is a try. But I changed my mind and your proposal seems to be more logical :) first execution is a regular execution. And overall (maxim) number of possible invocations will be retries + 1

Reply:
fixed